src/controllers/bookController.ts

```diff
@@ -1,29 +1,151 @@
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            let id = Number(req.params.id);
+            if (isNaN(id)) {
+                return res.status(400).json({ status: 'Invalid Id' });
+            } else {
+                const sqlRequest = `SELECT * FROM books WHERE id = ${id}`;
+                const databaseRequest = new TediousRequest(
+                    sqlRequest,
+                    function (err, _rowCount, rows) {
+                        if (err) {
+                            console.log(err);
+                            throw err;
+                        } else {
+                            if (rows.length == 0) {
+                                return res
+                                    .status(400)
+                                    .json({ status: 'book not found' });
+                            } else {
+                                for (let j = 0; j < rows[0].length; j++) {
+                                    output.push(rows[0][j].value);
+                                }
+                                return res.status(200).json(output);
+                            }
+                        }
+                    },
+                );
+                requestStatement = connection.execSql(databaseRequest);
+            }
+        });
+        connection.connect();
     }
 
-    getBook(req: Request, res: Response) {
-        // TODO: implement functionality
-        return res.status(500).json({
-            error: 'server_error',
-            error_description: 'Endpoint not implemented yet.',
+    async createBook(req: Request, res: Response) {
+        const connection = new Connection(this.config);
+        let requestStatement = new String(); //Plz ignore
+        const output = [];
+
+        // Setup event handler when the connection is established.
+        connection.on('connect', async function (err) {
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            const parsedBookTitle = req.params.bookTitle.split('_').join(' ');
+
+            const sqlRequest = `INSERT INTO books (BookTitle, Copies, ISBN) 
+                VALUES ( '${parsedBookTitle}', 
+                ${req.params.copies},
+                ${req.params.ISBN}) 
+                INSERT INTO authors (FirstName, Surname) 
+                VALUES ( '${req.params.authorFirst}', 
+                '${req.params.authorLast}') 
+                INSERT INTO books_authors (Author_Id, Book_Id) 
+                VALUES ( IDENT_CURRENT('authors'), IDENT_CURRENT('books')) `;
+
+            console.log(sqlRequest);
+
+            const databaseRequest = new TediousRequest(sqlRequest, function (
+                err,
+            ) {
+                if (err) {
+                    console.log(err);
+                } else {
+                    return res
+                        .status(200)
```

Comment:
It would be more typical to return a 201 without any body data for something like this, so that the client knows that something has been created.